pennylane_lightning/core/src/simulators/lightning_kokkos/StateVectorKokkos.hpp

```diff
@@ -500,14 +583,34 @@ class StateVectorKokkos final
         applyMultiQubitOp(matrix_, wires, inverse);
     }
 
+    /**
+     * @brief Apply a given matrix directly to the statevector using a
+     * raw matrix pointer vector.
```

Comment:
Does it only work for the pointer to the host memory?

Reply:
Yes, this matrix pointer is for host memory